lib/widget/listwidget.h

```diff
@@ -47,9 +47,17 @@ struct TabSelectionStyle
 
 class TabSelectionWidget : public WIDGET
 {
+protected:
+	TabSelectionWidget(): WIDGET(), currentTab(0), tabsAtOnce(1) {}
+	virtual void initialize();
 
 public:
-	TabSelectionWidget(WIDGET *parent);
+	static std::shared_ptr<TabSelectionWidget> make()
+	{
+		auto widget = std::shared_ptr<TabSelectionWidget>(new TabSelectionWidget());
```

Comment:
```suggestion
		auto widget = std::make_shared<TabSelectionWidget>();
```

Reply:
In order to use `make_shared` I would need to make the constructor public. :slightly_frowning_face: 

That's why I chose to use `shared_ptr` instead. I think it is the least undesirable option.

Reply:
> In order to use `make_shared` I would need to make the constructor public. 🙁
> 
> That's why I chose to use `shared_ptr` instead. I think it is the least undesirable option.

Ah - I blanked on that restriction. Hmm.

Reply:
As we discussed, fixed using local class that extends from the WIDGET.